test/common/http/filter/ip_tagging_filter_test.cc

```diff
@@ -65,32 +49,246 @@ class IpTaggingFilterTest : public testing::Test {
+  TestHeaderMapImpl request_headers;
+
+  EXPECT_CALL(stats_, counter("prefix.ip_tagging.total")).Times(1);
+  EXPECT_CALL(stats_, counter("prefix.ip_tagging.external_request.hit")).Times(1);
+
+  Network::Address::InstanceConstSharedPtr remote_address =
+      Network::Utility::parseInternetAddress("1.2.3.4");
+  EXPECT_CALL(filter_callbacks_.request_info_, downstreamRemoteAddress())
+      .WillOnce(ReturnRef(remote_address));
+
+  EXPECT_EQ(FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, false));
+  EXPECT_EQ("external_request", request_headers.get_(Headers::get().EnvoyIpTags));
 
-  EXPECT_EQ(FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers_, false));
   EXPECT_EQ(FilterDataStatus::Continue, filter_->decodeData(data_, false));
-  EXPECT_EQ(FilterTrailersStatus::Continue, filter_->decodeTrailers(request_headers_));
+  EXPECT_EQ(FilterTrailersStatus::Continue, filter_->decodeTrailers(request_headers));
+
+  // Check internal requests don't get a tag.
+  request_headers = {{"x-envoy-internal", "true"}};
+  EXPECT_EQ(FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, false));
+  EXPECT_FALSE(request_headers.has(Headers::get().EnvoyIpTags));
 }
 
 TEST_F(IpTaggingFilterTest, BothRequest) {
-  SetUpTest(both_request_json);
+  std::string both_request_yaml = R"EOF(
+request_type: both
+ip_tags:
+  - ip_tag_name: external_request
+    ip_list:
+      - {address_prefix: 1.2.3.4, prefix_len: 32}
+  - ip_tag_name: internal_request
+    ip_list:
+      - {address_prefix: 1.2.3.5, prefix_len: 32}
+)EOF";
+
+  SetUpTest(both_request_yaml);
+  EXPECT_EQ(FilterRequestType::BOTH, config_->requestType());
+  TestHeaderMapImpl request_headers{{"x-envoy-internal", "true"}};
+  EXPECT_CALL(stats_, counter("prefix.ip_tagging.total")).Times(2);
+  EXPECT_CALL(stats_, counter("prefix.ip_tagging.internal_request.hit")).Times(1);
+  EXPECT_CALL(stats_, counter("prefix.ip_tagging.external_request.hit")).Times(1);
+
+  Network::Address::InstanceConstSharedPtr remote_address =
+      Network::Utility::parseInternetAddress("1.2.3.5");
+  EXPECT_CALL(filter_callbacks_.request_info_, downstreamRemoteAddress())
+      .WillOnce(ReturnRef(remote_address));
+
+  EXPECT_EQ(FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, false));
+  EXPECT_EQ("internal_request", request_headers.get_(Headers::get().EnvoyIpTags));
+
+  request_headers = {};
+  remote_address = Network::Utility::parseInternetAddress("1.2.3.4");
+  EXPECT_CALL(filter_callbacks_.request_info_, downstreamRemoteAddress())
+      .WillOnce(ReturnRef(remote_address));
+
+  EXPECT_EQ(FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, false));
+  EXPECT_EQ("external_request", request_headers.get_(Headers::get().EnvoyIpTags));
+}
+
+TEST_F(IpTaggingFilterTest, NoHits) {
+  SetUpTest(internal_request_yaml);
+  TestHeaderMapImpl request_headers{{"x-envoy-internal", "true"}};
```

Comment:
Nit: a bunch of the locals in this file can be `const`.

Reply:
I const'ed(new verb?) the remaining strings. TestHeaderMapImpls can't be const; they are intended to be manipulated by the filter.